File: two_pointer_algorithms.py
class Solution:
    def twoSum(self, nums, target):
        # Using two pointer algorithms (assuming nums is a sorted array)
        N = len(nums)
        i, j = 0, N-1
        while i<j:
            current_sum = nums[i] + nums[j]
            if  current_sum < target:
                i += 1
            elif current_sum > target:
                j -= 1
            else:
               return [i, j]
                  

File: two_pointer_algorithms.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        // Using 2 pointer algorithms 
        int left, right;
        left = 0;
        right = (nums.Length) - 1;
        while (left < right){
            int current_sum = nums[left] + nums[right];
            if (current_sum < target){
                left += 1;
            }
            else if (current_sum > target) {
                right -= 1;
            }
            else{
                int [] res = new int[]{left, right};
                return res;
            }
        }
        return new int []{-1, -1};
    }
}
